### PR TITLE
[CALCITE] Support for attributes in alternative type conversion syntax

### PR DESCRIPTION
**Pull request by michaelvdow**
- Data attribute support is added for alternative type conversion syntax and explicit CAST calls
 - Removed SqlFormatFunction since this functionality is now supported by the data attributes for alternative type conversion syntax

### CODE REVIEW

File: parsing/dialects/dialect1/config.fmpp [hunk 1]
```diff
@@ -1123,15 +1124,13 @@ data: {
       "InlineCaseSpecific"
       "InlineCaseSpecificNamedFunctionCall"
       "InlineModOperator"
-      "AlternativeTypeConversionLiteralOrIdentifier"
-      "InlineFormatLiteralOrIdentifier"
       "NamedLiteralOrIdentifier"
+      "AlternativeTypeConversionLiteralOrIdentifier"
```
Comment (dasch-1) [inline parsing/dialects/dialect1/config.fmpp:1128]: nit: Can we order alphabetically? If not, we should probably add a comment saying why.

File: parsing/dialects/dialect1/config.fmpp [hunk 2]
```diff
@@ -1123,15 +1124,13 @@ data: {
...
     ]
 
     postExpressionMethods: [
-      "AlternativeTypeConversionQuery"
-      "InlineFormatQuery"
       "NamedQuery"
+      "AlternativeTypeConversionQuery"
```
Comment (dasch-1) [inline parsing/dialects/dialect1/config.fmpp:1133]: nit: Can we order alphabetically? If not, we should probably add a comment saying why.

File: parsing/dialects/dialect1/parserImpls.ftl [hunk 3]
```diff
@@ -1474,50 +1486,66 @@ SqlNode AlternativeTypeConversionLiteralOrIdentifier() :
...
+            <INTERVAL> interval = IntervalQualifier() { args.add(interval); }
+        )
+        [ <COMMA> AlternativeTypeConversionAttributeList(args) ]
+    |
+        AlternativeTypeConversionAttributeList(args)
+        [
+            (
+                <COMMA> dt = DataTypeAlternativeCastSyntax() { args.add(1, dt); }
+            |
+                <COMMA> <INTERVAL> interval = IntervalQualifier() { args.add(1, interval); }
```
Comment (dasch-1) [inline parsing/dialects/dialect1/parserImpls.ftl:1542]: nit: Line wrap (here and other lines in this function)

File: parsing/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ParserTest.java [hunk 4]
```diff
@@ -1746,40 +1746,40 @@ private void checkExNotNull(SqlParserUtil.StringAndPos sap, Throwable thrown) {
     sql(sql).ok(expected);
   }
 
-  @Test void testInlineFormatSimpleIdentifier() {
+  @Test void testAlternativeCastAttributeSimpleIdentifier() {
```
Comment (dasch-1) [inline parsing/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ParserTest.java:1749]: nit: Should we have the word "format" somewhere in this test name? (Here and other functions below).

File: parsing/dialects/dialect1/config.fmpp (hunk 1 above)
Comment (michaelvdow) [reply parsing/dialects/dialect1/config.fmpp:1128]: There's some tests that fail if the Named method comes after. I can write a comment to explain why

File: parsing/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ParserTest.java (hunk 4 above)
Comment (michaelvdow) [reply parsing/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ParserTest.java:1749]: Sure I'll update them with Format

**Pull request merged**